feat: Update fmha kernels to support Eagle3 on blackwell.

PerkzZheng opened the pull request.

PerkzZheng commented:
/bot run

PerkzZheng commented:
/bot run

tensorrt-cicd commented:
[PR_Github #1756](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1756/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1757](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1757/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1756](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1756/) [ run ] completed with state `ABORTED`

tensorrt-cicd commented:
[PR_Github #1757](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1757/) [ run ] completed with state `FAILURE`

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #1782](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1782/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1782](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1782/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #1321](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/1321/) completed with status: 'FAILURE'

PerkzZheng committed:
update cubins

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

PerkzZheng committed:
enable eagle3 generation fmha kernels and fix several other issues

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

fix the failing tests

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

use eagle3 extend_ctx not on blackwell

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

fix eagle3 rope kernel

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #1856](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1856/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #1856](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/1856/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #1375](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/1375/) completed with status: 'FAILURE'

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #2086](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/2086/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #2086](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/2086/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #1529](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/1529/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #2189](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/2189/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #2189](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/2189/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #1582](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/1582/) completed with status: 'FAILURE'

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #2897](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/2897/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #2897](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/2897/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #2042](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/2042/) completed with status: 'FAILURE'

PerkzZheng commented:
/bot run --disable-fail-fast

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #3056](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/3056/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #3056](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/3056/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #2136](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/2136/) completed with status: 'FAILURE'

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #3115](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/3115/) [ run ] triggered by Bot

PerkzZheng commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #3124](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/3124/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #3124](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/3124/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #2175](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/2175/) completed with status: 'SUCCESS'

PerkzZheng commented:
@lfr-0531 @mikeiovine can you help review this MR (it is mainly the same as the internal ones, but fixes several failing tests). For your convenience, you can just review this [commit](https://github.com/NVIDIA/TensorRT-LLM/pull/3446/commits/b6117bcafab62099f8022e709b5ade4609e2e331) which has excluded the cubins.
